Editorial suggestions from Benjamin Kaduk's IESG review (#365)
MikeBishop/dns-alt-svc: draft-ietf-dnsop-svcb-https.md

RRSets are explicitly unordered collections, so the
SvcPriority field is used to impose an ordering on SVCB RRs.

> SVCB RRs with a smaller SvcPriority value SHOULD be given

Reviewer: ```suggestion
ServiceMode RRs with a smaller SvcPriority value SHOULD be given
```

preference over RRs with a larger SvcPriority value (the previous paragraph

> notwithstanding).

Author: My original thinking seems lost to the depths of time.
That said, it looks like both the "SHOULD have the same Mode" and "if AliasMode exists, MUST ignore ServiceMode" directives are potentially relevant here.  That is, the advice here remains good general advice, but is not grounds to ignore the specific directives already given.  In particular, ignoring this SHOULD (which is permitted) does not allow us to violate the previous MUST.

> notwithstanding).

Reviewer: Does the proposed change above solve that problem?

> preference over RRs with a larger SvcPriority value (the previous paragraph
> notwithstanding).

Reviewer 2: Incorporated Ben's proposed change.  With that can we remove the "(the previous paragraph notwithstanding)."

```suggestion
preference over RRs with a larger SvcPriority value.
```

> notwithstanding).

Author: Apparently I missed this when it came in, sorry.  Yes, the now-merged fix is a good fix -- thanks!

The presentation `value` SHALL be a comma-separated list ({{value-list}})
of one or more IP addresses of the appropriate
family in standard textual format {{!RFC5952}}, {{!RFC4001}}.  To enable simpler parsing,
this SvcParamValue MUST NOT contain escape sequences.

> The wire format for each parameter is a sequence of (the respective
> address-family) IP addresses in network
> byte order.  Like an A or AAAA RRSet, the list of addresses represents an

Reviewer 2: ```suggestion
The wire format for each parameter is a sequence of IP addresses in network
byte order (for the respective address-family).  
Like an A or AAAA RRSet, the list of addresses represents an
```

Moving the parenthetical to the end seems more readable?

The presentation `value` SHALL be a comma-separated list ({{value-list}})
of one or more IP addresses of the appropriate

> family in standard textual format {{!RFC5952}}, {{!RFC4001}}.  To enable simpler parsing,

Author: 5952 only covers IPv6 textual format.

"mandatory" is used to indicate any mandatory keys for this RR, in addition to
any automatically mandatory keys that are present.

> A ServiceMode RR is considered "compatible" by a client if the client

Author: This one I'm not entirely sure about.
It is perhaps possible that it makes sense to think about a RR being "compatible with" a client, if the RR itself is the focus of the concept.  Yet the rest of the discussion here seems focused on client processing, and the phrasing is a little weird to me anyway, so I think this is probably better.

Reviewer 2: I can also see this going either way.

Reviewer: "compatible with" seems more natural to me, but on matters of comprehensibility I defer to the reader.

    ech=... key65333=ex1 key65444=ex2 mandatory=key65444,ech

In wire format, the keys are represented by their numeric values in

> network byte order, concatenated in strictly increasing numeric order.

Author: we used the phrase "strictly increasing numeric order" earlier, so this seeks consistency.

Use of any protocol with SVCB requires a protocol-specific mapping
specification.  This section specifies the mapping for the "http" and "https"

> URI schemes {{!HTTP}}.

Author: currently we list [HTTP] and [I-D.ietf-httpbis-semantics] as different reference entries.  I don't have a build toolchain set up, though, so I'm not 100% sure this is a complete fix.

## Interaction with Alt-Svc

> Clients that implement support for both Alt-Svc and HTTPS records and
> receive an Alt-Svc header field in a response SHOULD

Reviewer 2: ```suggestion
Clients that implement support for both Alt-Svc and HTTPS records and
are making a connection based on a cached Alt-Svc response SHOULD
```

Alt-Svc can come from an H2 frame not just a header.
The lookup also should come at request time, not at the time of receiving the Alt-Svc,
as these could be separated by hours or days.

    @ 7200 IN HTTPS 1 . alpn=h3

to indicate that https://simple.example supports QUIC

> in addition to HTTP/1.1 over TLS over TCP (the implicit default).

Reviewer 2: ```suggestion
in addition to HTTP over TLS over TCP (the implicit default,
with HTTP/1.1 or HTTP/2 negotiated during the TLS handshake 
via ALPN).
```

It could be either as ALPN will negotiate.

Reviewer: Yeah, but the sentence is: "The domain owner could add this record ... to indicate that it ... supports ...".  The record indicates support for HTTP/3 and HTTP/1.1, not HTTP/2.  An HTTP/2-only would not attempt to use this record, because there is no indication that the endpoint supports HTTP/2.

# Security Considerations

> SVCB/HTTPS RRs permit distribution over untrusted

Author: just a couple paragraphs prior we said "intended for use over a privacy-preserving DNS transport", which is not exactly an untrusted channel (though it's also not a fully trusted channel).

Reviewer 2: makes sense

    \x02\x9b                                           # key 667
    \x00\x09                                           # length 9
    hello\xd2qoo                                       # value

> {: title="A generic key and quoted value with an octal escape"}

Author: I'm not sure what "decimal escape" was intended to mean; this is a best guess

Reviewer: Nope, it's actually a decimal!  RFC 1035 Section 5.1:

~~~
\DDD            where each D is a digit is the octet corresponding to
                the decimal number described by DDD.  The resulting
                octet is assumed to be text and is not checked for
                special meaning.
~~~

Zone files actually don't support hex, octal, or unicode escaping.

Author: Oops, reverted.
I ... am not really sure why my brain saw 1-255 and somehow got "octal" from that :-/

Reviewer 3: I'd guess "resulting octet" got slurred in "resulting octal value."
